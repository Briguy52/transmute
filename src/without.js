import curry from "./curry";
import { Seq } from "immutable";

/**
 * Removes items in `unwanted` from `subject`.
 *
 * @param  {Iterable} unwanted
 * @param  {Iterable} subject
 * @return {Iterable}
 */
function without(unwanted, subject) {
  unwanted = Seq.Set(unwanted);
  return subject.filterNot(value => unwanted.includes(value));
}

export default curry(without);
