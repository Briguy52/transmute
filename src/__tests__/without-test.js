import { List } from "immutable";
import without from "../without";

describe("transmute/without", () => {
  it("drops unwanted values", () => {
    expect(
      without(["one", "three"], List.of("one", "two", "three"))
    ).toMatchSnapshot();
  });
});
