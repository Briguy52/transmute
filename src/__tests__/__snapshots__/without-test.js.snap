// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/without drops unwanted values 1`] = `
Immutable.List [
  "two",
]
`;
